### DISCUSSION TITLE
How to add context menu items from the specification editor

### DISCUSSION THREAD

**Discussion by t-chiang** [eclipse-sirius/sirius-desktop#476] [Q&A] [answered]
Hello all,

I want to add menu items using the viewpoint specification editor without having to get into Java external call. The operations I want to do are fairly simple and would be much easier to implement using Sirius instead of getting into Java.

I've come across examples here: https://github.com/eclipse-sirius/sirius-desktop/tree/master/plugins/org.eclipse.sirius.tests.swtbot/data/unit/popupMenu 

However I am unable to view new context menu options when copying what they've done here. So far I'm stuck with the following that aren't working:
![image](https://github.com/user-attachments/assets/44d0e0a7-c114-411e-9770-da862df15be5)
 
![image](https://github.com/user-attachments/assets/5c4cc053-e970-4a91-b36d-8f9124a2f518)

Any help would be appreciated for getting these context menu options working!

**Comment by t-chiang** [comment, marked as answer]
The answer was to add the org.eclipse.ui.menus to the extensions of my project and reference in as menu like it is in the examples.

![image](https://github.com/user-attachments/assets/b3b21f62-d011-4838-81b7-64f2945a224c)

**Comment by lredor** [comment]
Hi,
it seems to be the same question as https://github.com/eclipse-sirius/sirius-desktop/discussions/453. I proposed an example on the other discussion. It seems strange to have to add the extension point in your project without actually really using it. But all the better if it works.
Can you also close the discussion 453 if the problem is the same ?

**Reply by t-chiang** [reply]
Hi Laurent,

I'm unable to close https://github.com/eclipse-sirius/sirius-desktop/discussions/453 as it wasn't my post originally. I do appreciate the answers you posted there nonetheless.

Thanks
Tom
